To: vim_dev@googlegroups.com
Subject: Patch 7.3.348
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.348
Problem:    "call range(1, 947948399)" causes a crash. (ZyX)
Solution:   Avoid a loop in the out of memory message.
Files:	    src/misc2.c


*** ../vim-7.3.347/src/misc2.c	2011-08-19 22:28:58.000000000 +0200
--- src/misc2.c	2011-10-26 11:17:26.000000000 +0200
***************
*** 1012,1019 ****
      {
  	/* Don't hide this message */
  	emsg_silent = 0;
! 	EMSGN(_("E342: Out of memory!  (allocating %lu bytes)"), size);
  	did_outofmem_msg = TRUE;
      }
  }
  
--- 1012,1023 ----
      {
  	/* Don't hide this message */
  	emsg_silent = 0;
! 
! 	/* Must come first to avoid coming back here when printing the error
! 	 * message fails, e.g. when setting v:errmsg. */
  	did_outofmem_msg = TRUE;
+ 
+ 	EMSGN(_("E342: Out of memory!  (allocating %lu bytes)"), size);
      }
  }
  
*** ../vim-7.3.347/src/version.c	2011-10-26 11:36:21.000000000 +0200
--- src/version.c	2011-10-26 11:38:33.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     348,
  /**/

-- 
A fool learns from his mistakes, a wise man from someone else's.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
